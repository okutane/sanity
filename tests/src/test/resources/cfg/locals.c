static void localUndefined() {
    int i;
}

static void localLateDefinition() {
    int i;
    i = 1;
}

static void localInstaDefinition() {
    int i = 1;
}

static void addressOf() {
    int i;
    int *pi = &i;
}
