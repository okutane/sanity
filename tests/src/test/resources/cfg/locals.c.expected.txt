CFG: localUndefined
#0000:	 alloc: i
 - no source -
#0001:	 return <exit>
locals.c:3 }

CFG: localLateDefinition
#0000:	 alloc: i
 - no source -
#0001:	 assign: *i = 1
locals.c:7 i = 1;
#0002:	 return <exit>
locals.c:8 }

CFG: localInstaDefinition
#0000:	 alloc: i
 - no source -
#0001:	 assign: *i = 1
locals.c:11 int i = 1;
#0002:	 return <exit>
locals.c:12 }

CFG: addressOf
#0000:	 alloc: i
 - no source -
#0001:	 alloc: pi
 - no source -
#0002:	 assign: *pi = i
locals.c:16 int *pi = &i;
#0003:	 return <exit>
locals.c:17 }

